src/tracing/oltp_tracing.rs

```diff
@@ -0,0 +1,127 @@
+use opentelemetry::{
+    runtime::Tokio,
+    sdk::{trace::config, Resource},
+    KeyValue,
+};
+use opentelemetry_otlp::WithExportConfig;
+use pyo3::{exceptions::PyValueError, pyclass, pymethods, PyAny, PyResult};
+use tracing_subscriber::{layer::SubscriberExt, EnvFilter, Layer, Registry};
+
+use super::{log_layer, TracingConfig, TracingSetupError};
+
+/// Send traces to the opentelemetry collector:
+/// https://opentelemetry.io/docs/collector/
+///
+/// This is the recommended approach since it allows
+/// the maximum flexibility in what to do with all the data
+/// bytewax can generate.
+#[pyclass(module="bytewax.tracing", extends=TracingConfig)]
+#[pyo3(text_signature = "(service_name, url, protocol)")]
+#[derive(Clone)]
+pub(crate) struct OltpTracingConfig {
+    /// Service name, identifies this dataflow.
+    #[pyo3(get)]
+    pub(crate) service_name: String,
+    /// Optional collector's URL
+    #[pyo3(get)]
+    pub(crate) url: Option<String>,
+    /// Optional protocol. This can be either:
+    /// - "GRPC" (default)
+    /// - "HTTP"
+    #[pyo3(get)]
+    pub(crate) protocol: Option<String>,
+}
+
+impl OltpTracingConfig {
+    pub(crate) fn setup(self) -> Result<(), TracingSetupError> {
+        // Instantiate the builder
+        let mut exporter = opentelemetry_otlp::new_exporter().tonic();
+
+        // Change the protocol if required
+        if let Some(protocol) = self.protocol {
+            exporter = match protocol.as_str() {
+                "HTTP" => exporter.with_protocol(opentelemetry_otlp::Protocol::HttpBinary),
+                "GRPC" => exporter.with_protocol(opentelemetry_otlp::Protocol::Grpc),
+                val => panic!("Unknown protocol: {val}. Only 'GRPC' and 'HTTP' allowed"),
+            }
+        }
+
+        // Change the url if required
+        if let Some(endpoint) = self.url {
+            exporter = exporter.with_endpoint(endpoint);
+        }
+
+        // Create the tracer
+        let tracer = opentelemetry_otlp::new_pipeline()
```

Comment:
mmh, I assumed sampling would be configured in the otlp itself, but I still need to check this is supported.

Reply:
Sampling is supported by the OLTP collector itself, but it might make sense to sample data before sending to it. I added a config option, by default sampling ratio is 1 (all traces are sampled).